tests/filecheck/transforms/memref_streamify.mlir

```diff
@@ -1,5 +1,9 @@
 // RUN: xdsl-opt -p memref-streamify %s | filecheck %s
 
+// Check that streamfying twice does not make further changes
+// RUN: xdsl-opt -p memref-streamify,memref-streamify %s | filecheck %s
```

Comment:
This is a very neat idea! I like adding these kinds of things to tests!

Reply:
idempotency is a pre-requisite for our exhaustive search of IRs not to explode BTW